crates/runtime/src/embeddings/metrics.rs

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2024-2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use std::{sync::LazyLock, time::Duration};
+
+use async_openai::types::{CreateEmbeddingRequest, EncodingFormat};
+use opentelemetry::{
+    global,
+    metrics::{Counter, Histogram, Meter},
+    Key, KeyValue, Value,
+};
+
+static METER: LazyLock<Meter> = LazyLock::new(|| global::meter("embedding"));
+
+pub(crate) static EMBEDDING_REQUESTS: LazyLock<Counter<u64>> =
+    LazyLock::new(|| METER.u64_counter("embedding_requests").build());
+
+pub(crate) static FAILURES: LazyLock<Counter<u64>> = LazyLock::new(|| {
+    METER
+        .u64_counter("embedding_failures")
+        .with_description("Number of embedding failures.")
+        .build()
+});
+
+pub(crate) static EMBEDDING_INTERNAL_DURATION_MS: LazyLock<Histogram<f64>> = LazyLock::new(|| {
+    METER
+        .f64_histogram("embedding_internal_request_duration_ms")
```

Comment:
```suggestion
        .f64_histogram("embeddings_internal_request_duration_ms")
```

To be consistent with `embeddings_load_errors`/etc where `embeddings` is in plural

Reply:
I do think we should possibly change `embeddings_load_errors` to `embedding_load_errors`. But since we already have it as a plural I will follow convention. We can change all afterwards.